### ISSUE TITLE
Bad multi-level interrupt encoding for nodes consuming interrupts from multiple aggregators

### ISSUE THREAD

**Issue by LaurentiuM1234** [zephyrproject-rtos/zephyr#69820]
**Describe the bug**
When dealing with a node consuming interrupts from multiple aggregators extending different interrupt lines,
the interrupt IDs will not be encoded properly.

**To Reproduce**
Any setup such as the following will do:

```
intc1: interrupt-controller@cafebabe {
    /* some more properties here */
    interrupt-controller;
    #interrupt-cells = <1>;
};

intc2: interrupt-controller@deadbeef {
     /* some more properties here */
    interrupt-controller;
    interrupts-extended = <&intc1 14>;
};

intc3: interruptc-controller@babebeef {
     /* some more properties here */
    interrupt-controller;
    interrupts-extended = <&intc1 15>;
};

consumer_node: ip@cafecafe {
    /* some more properties here */
    interrupts-extended = <&intc2 16>, <&intc3 15>;
};
```

The interrupts from `consumer_node` will be encoded as follows: `14 | ((16 + 1) << 8)` and `14 | ((15 + 1) << 8)`

**Expected behavior**
The L1 interrupts should be 14 and 15 instead of 14 and 14.

**Fix**
The issue is fixed by https://github.com/zephyrproject-rtos/zephyr/pull/68784.

**Comment by aescolar** [comment]
CC @ycsin

**Comment by ycsin** [comment]
fixed by #68784
